nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -76,7 +81,25 @@ WaypointFollower::on_configure(const rclcpp_lifecycle::State & /*state*/)
     get_node_clock_interface(),
     get_node_logging_interface(),
     get_node_waitables_interface(),
-    "follow_waypoints", std::bind(&WaypointFollower::followWaypoints, this));
+    "follow_waypoints", std::bind(&WaypointFollower::followWaypointsCallback, this));
+
+  from_ll_to_map_client_ = std::make_unique<
+    nav2_util::ServiceClient<robot_localization::srv::FromLL,
+    std::shared_ptr<nav2_util::LifecycleNode>>>(
+    "/fromLL",
+    node);
+
+  gps_action_server_ = std::make_unique<ActionServerGPS>(
+    get_node_base_interface(),
+    get_node_clock_interface(),
+    get_node_logging_interface(),
+    get_node_waitables_interface(),
+    "follow_gps_waypoints",
+    std::bind(&WaypointFollower::followGPSWaypointsCallback, this));
+
+  // used for transfroming orientation of GPS poses to map frame
+  tf_buffer_ = std::make_shared<tf2_ros::Buffer>(node->get_clock());
```

Comment:
I think there's the timer interface that needs to be added? Unless something's changed in TF2 in the last couple distributions making that no longer strictly required?

Reply:
As far as I know, you are expected to pass a pointer to a clock object owned by the node when building the tf2_ros::Buffer object. This is how its done in the [docs](https://docs.ros.org/en/humble/Tutorials/Intermediate/Tf2/Writing-A-Tf2-Listener-Cpp.html) and is also all over the place in nav2 (not sure if that's what you are refering to). Anyway, I think these objects are not longer used (they were when I was confused and thought the transform between `map` and `utm` was needed. I will remove them as well